torch/jit/__init__.py

```diff
@@ -155,9 +130,43 @@ def __call__(cls, *args, **kwargs):
             # user passed in), this problem goes away, because the class
             # __init__ is a part of is indeed Foo.
 
+            old_init = arg.__init__
+            # Python 2 has a concept of unbound methods, which doesn't play well
+            # with what we want to do here. Therefore, unpack the method into a
```

Comment:
'what we want to do here' <- can you clarify?

Reply:
Basically the problem is that unbound methods are like functions, but with type checks on the first argument. If we didn't unwrap them you'll get errors like "__init__ expected the first argument to be of type MyModule, but it is MyModule". That's because it still checks for the old class, but it gets the decorated class. Unwrapping the method into a plain function gets rid of the type check. Python 3 already returns a plain function, so there's no need to do anything

Reply:
I see, that makes more sense.

Reply:
I'll rephrase it when I'll be pushing other fixes